src/librustc_mir/interpret/place.rs

```diff
@@ -1038,23 +1038,26 @@ where
         variant_index: VariantIdx,
         dest: PlaceTy<'tcx, M::PointerTag>,
     ) -> InterpResult<'tcx> {
-        let variant_scalar = Scalar::from_u32(variant_index.as_u32()).into();
+
+        // Layout computation excludes uninhabited variants from consideration
+        // so this checks that the `variant_index` is valid.
```

Comment:
The key point is that there's no alternative: there's no way to represent those variants in the given layout.

Reply:
I feel like there's been a lot of muddying the waters on "`variant_index` validity".

It's always valid, it's always in range, it's impossible for it to not be a variant (despite what has been said wrt the comments lower down).
Also, for `Multiple`, the `variants` vec *always* includes *all* variants declared in the `enum`.

What needs to be handled *here* is "the variant is uninhabited" i.e. *unrepresented*.
This means *two* things, not just one (another source of confusion):
* there is no tag/niche allocated for it, so the best you could do is write undef over the tag/niche
* fully initializing (which is what `SetDiscriminant` means) an uninhabited variant should never be reachable, as another uninhabited `Operand` *must've been* evaluated for *at least* one of the fields, so this should be some sort of error (I think @RalfJung is right that this is technically an "invalid program" situation)

Reply:
(of course the PR was updated while I was trying to post that comment, heh)

I think the new form of the comment seems fine, but I'd wait until @RalfJung also agrees, to proceed.